### PR TITLE
 RED-74 - Join Protocol: Reject nodes that are in debug mode

### PR DESCRIPTION
**Pull request by magonjr**
During the joining process, implement a way for the network to check if the joining node is in debug mode and reject it.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -6396,6 +6398,17 @@ const shardusSetup = (): void => {
           shardus.shutdownFromDapp(tag, message, false)
           return false
         }
+
+        //error out nodes in debug mode for production networks to prevent joining
+        if (
+          networkAccount.data.mode === ServerMode.Release &&
```
Comment (ahmxdiqbal) [inline src/index.ts:6771]: in the above else condition to fetch the network account from the archiver, will this network account also have a `mode` field?

Comment (magonjr) [reply src/index.ts:6771]: yes, tested, I didn't even need to add it to the archiver because _data_ is a _any_ type,

Comment (ahmxdiqbal) [reply src/index.ts:6771]: so was the mode field already part of the networkAccount we get from the archiver?

Comment (magonjr) [reply src/index.ts:6771]: I meant, once I added it to the Network account obj here (https://github.com/shardeum/shardeum/pull/42/files#diff-a9f7b5615265a25a5c043bea768b8d1fa86baf642022b1b7324dba2708145d7cR19) it was stored, picked up by the archiver and appeared on the archiver response as expected

Comment (ahmxdiqbal) [reply src/index.ts:6771]: ok makes sense

**Review by ahmxdiqbal — DISMISSED** [review]

**Pull request merged**